src/coreclr/vm/eventing/eventpipe/ep-rt-coreclr.h

```diff
@@ -1137,9 +1137,16 @@ ep_rt_entrypoint_assembly_name_get_utf8 (void)
 
 	// get the name from the host if we can't get assembly info, e.g., if the runtime is
 	// suspended before an assembly is loaded.
-	SString assembly_name;
-	if (HostInformation::GetProperty (HOST_PROPERTY_ENTRY_ASSEMBLY_NAME, assembly_name))
+	static bool read_name_from_host = false;
+	static SString assembly_name;
```

Comment:
This is going to have destructor that is going to run during shutdown. Destructors running during shutdown are well-known bug farm.

Reply:
Agreed. This looks like a case where we should just allocate some memory, store it in a global pointer, and let it leak. Updating a pointer atomically should be easy enough in this code path.

Reply:
I’ll change this to allocate a new copy of the string once and store it in a global that we let leak.